lib/array_equals.rb

```diff
@@ -3,3 +3,21 @@
 def array_equals(array1, array2)
   raise NotImplementedError
 end
+
+
+def array_equals(array1, array2)
+  if array1.nil? && array2.nil? || array1 == [] && array2 == []
+    return true
+  end
+
+  if array1 != array2
```

Comment:
Because you're not supposed to use != or == to compare the arrays in this exercise. 

```suggestion
  if array1.nil?  || array2.nil?
```